core/domain/email_jobs_one_off.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs for Sent Email Model."""
+
+from core import jobs
+from core.domain import email_manager
+from core.platform import models
+
+(email_models,) = models.Registry.import_models([models.NAMES.email])
+
+
+class SentEmailUpdateHashOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for updating hash for all Sent Email Models."""
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [email_models.SentEmailModel]
+
+    @staticmethod
+    def map(email_model):
+        if not email_model.email_hash or len(email_model.email_hash) == 0:
+            yield (email_model.id, None)
```

Comment:
Seems weird to do this and then fetch the instance again. Why not just do the update here (and have a no-op reduce() that's never called)?